crates/iceberg/src/spec/values.rs

```diff
@@ -0,0 +1,941 @@
+            ])),
+        );
+    }
+
+    #[test]
+    fn json_list() {
+        let record = r#"[1, 2, 3]"#;
+
+        check_json_serde(
+            record,
+            Literal::List(vec![
+                Some(Literal::Primitive(PrimitiveLiteral::Int(1))),
+                Some(Literal::Primitive(PrimitiveLiteral::Int(2))),
+                Some(Literal::Primitive(PrimitiveLiteral::Int(3))),
+            ]),
+            &Type::List(ListType {
+                element_field: NestedField {
+                    id: 0,
+                    name: "".to_string(),
+                    required: true,
+                    field_type: Box::new(Type::Primitive(PrimitiveType::Int)),
+                    doc: None,
+                    initial_default: None,
+                    write_default: None,
+                },
+            }),
+        );
+    }
+
+    #[test]
+    fn json_map() {
+        let record = r#"{ "keys": ["a", "b"], "values": [1, 2] }"#;
+
+        check_json_serde(
+            record,
+            Literal::Map(BTreeMap::from([
+                (
+                    Literal::Primitive(PrimitiveLiteral::String("a".to_string())),
+                    Some(Literal::Primitive(PrimitiveLiteral::Int(1))),
+                ),
+                (
+                    Literal::Primitive(PrimitiveLiteral::String("b".to_string())),
+                    Some(Literal::Primitive(PrimitiveLiteral::Int(2))),
+                ),
+            ])),
+            &Type::Map(MapType {
+                key_field: NestedField {
+                    id: 0,
+                    name: "key".to_string(),
+                    required: true,
+                    field_type: Box::new(Type::Primitive(PrimitiveType::String)),
+                    doc: None,
+                    initial_default: None,
+                    write_default: None,
+                },
+                value_field: NestedField {
+                    id: 1,
+                    name: "value".to_string(),
+                    required: true,
+                    field_type: Box::new(Type::Primitive(PrimitiveType::Int)),
+                    doc: None,
+                    initial_default: None,
+                    write_default: None,
+                },
+            }),
+        );
+    }
+
+    #[test]
+    fn avro_bytes_boolean() {
+        let bytes = vec![1u8];
+
+        check_avro_bytes_serde(
+            bytes,
+            Literal::Primitive(PrimitiveLiteral::Boolean(true)),
+            &Type::Primitive(PrimitiveType::Boolean),
+        );
+    }
+
+    #[test]
+    fn avro_bytes_int() {
+        let bytes = vec![32u8, 0u8, 0u8, 0u8];
+
+        check_avro_bytes_serde(
+            bytes,
+            Literal::Primitive(PrimitiveLiteral::Int(32)),
+            &Type::Primitive(PrimitiveType::Int),
+        );
+    }
+
+    #[test]
+    fn avro_bytes_long() {
+        let bytes = vec![32u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8];
+
+        check_avro_bytes_serde(
```

Comment:
What are we testing here? Keep in mind that Avro uses [zig-zag](https://en.wikipedia.org/wiki/Variable-length_quantity#Zigzag_encoding) encoding for integers.

Reply:
Typically the upper and lower bounds are stored as avro "bytes" inside the manifest_entry. These bytes conform to the iceberg [binary single value serialization](https://iceberg.apache.org/spec/#binary-single-value-serialization). This is a test if bytes stored serialized to avro "bytes" can be converted to a literal value